src/machine-learning/machine-learning-api-client.ts

```diff
@@ -37,17 +37,28 @@ export interface StatusErrorResponse {
 }
 
 /**
- * A Firebase ML Model input object
+ * Firebase ML Model input objects
  */
-export interface ModelOptions {
+export interface ModelOptionsBase {
   displayName?: string;
   tags?: string[];
-
-  tfliteModel?: { gcsTfliteUri: string };
 }
+export interface GcsTfliteModelOptions extends ModelOptionsBase {
+  tfliteModel: {
+    gcsTfliteUri: string;
+  };
+}
+export interface AutoMLTfliteModelOptions extends ModelOptionsBase {
+  tfliteModel: {
+    automlModel: string;
+  };
+}
+export type ModelOptions = ModelOptionsBase | GcsTfliteModelOptions | AutoMLTfliteModelOptions;
```

Comment:
Should ModelOptionsBase be included here? Is it ever used by itself except as one of the derived types?

Reply:
It is valid to specify a model with just displayName (and optional tags). So yes, it should be included.

Reply:
In that case why not just get rid of the union altogether, and use the type hierarchy in its place?

```
interface ModelOptions {

}

interface GcsTfliteModelOptions extends ModelOptions {

}
```

This avoids the `ModelOptionsBase` type. Any reason to not do this?

Reply:
The way it's currently implemented is what we agreed on in the design doc/CL that went with the design doc. I'm also pretty sure you are the one that came up with the 'type as a union' idea, and at the time we were discussing it a lot and coming up with how we wanted to implement it we both had a lot more recent context into why we might want to do it this way. (How would ModelUpdateOptions work with that structure?). I think we should leave it.